api/graphql/types/opening_hours/types.py

```diff
@@ -1,187 +1,58 @@
-    resource_state = graphene.String()
-    time_spans = graphene.List(TimeSpanType)
-    name_fi = graphene.String()
-    name_en = graphene.String()
-    name_sv = graphene.String()
-    description_fi = graphene.String()
-    description_en = graphene.String()
-    description_sv = graphene.String()
+    def resolve_end_datetime(self, info: GQLInfo):
+        return self.end_datetime
 
 
-class OpeningTimesType(graphene.ObjectType):
-    date = graphene.Date()
-    start_time = graphene.DateTime()
-    end_time = graphene.DateTime()
-    state = graphene.String()
-    periods = graphene.List(graphene.Int)
-    is_reservable = graphene.Boolean()
+class ReservableTimeSpansGraphQLMixin:
+    reservable_time_spans = graphene.List(
+        ReservableTimeSpanType,
+        start_date=graphene.Date(),
+        end_date=graphene.Date(),
+    )
 
-    def resolve_date(self, info):
-        return self.date
+    def resolve_reservable_time_spans(
+        self: ReservationUnit | Unit,
+        *_,
+        **kwargs,
+    ) -> list[ReservableTimeSpanType] | None:
+        start_date = kwargs.get("start_date")
+        end_date = kwargs.get("end_date")
 
-    def resolve_start_time(self, info):
-        if not self.start_time:
+        # All parameters are required to get reservable time spans
+        if not (start_date and end_date):
             return None
 
-        return self.start_time
+        origin_hauki_resource = self.origin_hauki_resource
 
-    def resolve_end_time(self, info):
-        if not self.end_time:
+        if not origin_hauki_resource:
             return None
 
-        return self.end_time
-
-    def resolve_periods(self, info, **kwargs):
-        return self.periods
-
-    def resolve_is_reservable(self, info, **kwargs):
-        return self.is_reservable
-
-
-class OpeningHoursType(graphene.ObjectType):
-    opening_times = graphene.List(OpeningTimesType)
-    opening_time_periods = graphene.List(PeriodType)
-
-
-class OpeningHoursMixin:
-    hauki_origin_id = settings.HAUKI_ORIGIN_ID
-
-    opening_hours = graphene.Field(
-        OpeningHoursType,
-        opening_times=graphene.Boolean(),
-        periods=graphene.Boolean(),
-        start_date=graphene.Date(),
-        end_date=graphene.Date(),
-    )
-
-    def resolve_opening_hours(self: ReservationUnit | Unit, *_, **kwargs):
-        start = kwargs.get("start_date")
-        end = kwargs.get("end_date")
-        init_periods = kwargs.get("periods", False)
-        init_times = kwargs.get("opening_times", False)
-        if not (start and end):
-            init_times = False
-
-        opening_hours_client = OpeningHoursClient(
-            resources=self.hauki_resource_origin_id,
-            start_date=start,
-            end_date=end,
-            init_periods=init_periods,
-            init_opening_hours=init_times,
-            hauki_origin_id=self.hauki_resource_data_source_id,
+        origin_hauki_resource.reservable_time_spans.filter(
```

Comment:
Unnecessary call? The next command uses `.filter_period` which uses `start_datetime__lt=end_date` & `end_datetime__gt=start_date`. There is a slight difference: `start_datetime__lte` vs `start_datetime__lt`. Intentional?